tool_watershed/watershed_analysis_networkx.py

```diff
@@ -36,6 +39,10 @@ def __init__(
         self._threshold = threshold
         self._aggregate = None
         self._graph = None
+        self.model = model
+        self.result_id = None  # Id of result_item refering to netcdf/gridadmin
+        if result_item:
+            self.result_id = result_item.id
```

Comment:
instead of line 43 to 45, ``self.result_id = result_item.id`` would achieve the exact same thing I think?

Reply:
I think this can be left out altogether; see my later comment suggesting a simpler implementation